command/server.go

```diff
@@ -1641,39 +1641,86 @@ func (c *ServerCommand) Run(args []string) int {
 			logWriter := c.logger.StandardWriter(&hclog.StandardLoggerOptions{})
 			pprof.Lookup("goroutine").WriteTo(logWriter, 2)
 
-			if os.Getenv("VAULT_STACKTRACE_WRITE_TO_FILE") != "" {
-				c.logger.Info("Writing stacktrace to file")
-
-				dir := ""
-				path := os.Getenv("VAULT_STACKTRACE_FILE_PATH")
-				if path != "" {
-					if _, err := os.Stat(path); err != nil {
-						c.logger.Error("Checking stacktrace path failed", "error", err)
-						continue
-					}
-					dir = path
-				} else {
-					dir, err = os.MkdirTemp("", "vault-stacktrace")
-					if err != nil {
-						c.logger.Error("Could not create temporary directory for stacktrace", "error", err)
-						continue
-					}
+			if os.Getenv("VAULT_PPROF_SIGUSR2_DUMP") != "" {
+				dir, err := os.MkdirTemp("", "pprof")
+				if err != nil {
+					c.UI.Error(err.Error())
+					continue
+				}
+
+				fGoRoutine, err := os.Create(fmt.Sprintf("%s/goroutines", dir))
```

Comment:
It looks like you're dropping the existing "stacktrace" output, which was the comprehensive text version of the stacks, and replacing it with this pprof version of the stacks, which may not include everything.  You're also removing the ability to specify a path for where the files should go. Could you retain the existing functionality please?